Mage.Sets/src/mage/cards/d/DreamDevourer.java

```diff
@@ -86,9 +84,28 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
         for (Card card : controller.getHand().getCards(filter, game)) {
-            String costText = CardUtil.reduceCost(card.getSpellAbility().getManaCostsToPay(), 2).getText();
-            game.getState().setValue(card.getId().toString() + "Foretell Cost", costText);
-            ForetellAbility foretellAbility = new ForetellAbility(card, costText);
+            ForetellAbility foretellAbility;
+            if (card instanceof SplitCard) {
+                String leftHalfCost = CardUtil.reduceCost(((SplitCard) card).getLeftHalfCard().getSpellAbility().getManaCostsToPay(), 2).getText();
```

Comment:
You must use `card.getManaCost()` not dynamic `getSpellAbility().getManaCostsToPay()` (the last one uses in resolve or other things to find real mana cost after all cost modification effects).